graph/path/morphism_apply_functions.go

```diff
@@ -20,58 +20,80 @@ import (
 	"github.com/google/cayley/quad"
 )
 
+// join puts two iterators together by intersecting their result sets with an AND
```

Comment:
I was waffling between the two-- on one hand, we're logically intersecting two result sets here. On the other, the intention is to join two paths together. In that sense, I think calling this 'join' is a lot more evocative of intent, if not as clear on mechanics.